test/test_table.py

```diff
@@ -0,0 +1,147 @@
+from hpack.table import HeaderTable
+import pytest
+import sys
+_ver = sys.version_info
+is_py2 = _ver[0] == 2
+is_py3 = _ver[0] == 3
+
+class TestHeaderTable(object):
+    def test_getitem_dynamic_table(self):
+        tbl = HeaderTable()
+        off = len(HeaderTable.STATIC_TABLE)
+        val = (b'TestName', b'TestValue')
+        tbl.add(val[0], val[1])
+        res = tbl[off + 1]
+        assert res == val
+
+    def test_getitem_static_table(self):
+        tbl = HeaderTable()
+        exp = HeaderTable.STATIC_TABLE[0]
+        res = tbl[1]
+        assert res == exp
+        off = len(HeaderTable.STATIC_TABLE)
+        exp = HeaderTable.STATIC_TABLE[off - 1]
+        res = tbl[off]
+        assert res == exp
+
+    def test_getitem_zero_index(self):
+        tbl = HeaderTable()
+        res = tbl[0]
+        assert res == None # TODO HPACKException will be raised instead
+
+    def test_getitem_out_of_range(self):
+        tbl = HeaderTable()
+        off = len(HeaderTable.STATIC_TABLE)
+        tbl.add(b'TestName', b'TestValue')
+        res = tbl[off+2]
+        assert res == None # TODO HPACKException will be reaised instead
+
+    def test_setitem(self):
+        tbl = HeaderTable()
+        with pytest.raises(TypeError) as einfo:
+            tbl[1] = (b'TestName', b'TestValue')
+        assert 'HeaderTable' in str(einfo.value)
+
+    def test_repr(self):
+        tbl = HeaderTable()
+        tbl.add(b'TestName1', b'TestValue1')
+        tbl.add(b'TestName2', b'TestValue2')
+        tbl.add(b'TestName2', b'TestValue2')
+        # Meh, I hate that I have to do this to test
+        # repr
+        if(is_py3):
+            exp = ("HeaderTable(4096, False, ["    +
+                   "(b'TestName2', b'TestValue2'), " +
+                   "(b'TestName2', b'TestValue2'), " +
+                   "(b'TestName1', b'TestValue1')"   +
+                   "])")
+        else:
+            exp = ("HeaderTable(4096, False, ["    +
+                   "('TestName2', 'TestValue2'), " +
+                   "('TestName2', 'TestValue2'), " +
+                   "('TestName1', 'TestValue1')"   +
+                   "])")
+        res = str(tbl)
```

Comment:
*Strictly* this is testing the cast to string, not the repr. ;) Might be better to call `repr()` than `str()`.